packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { RemoteParticipant } from '@azure/communication-calling';
+import { _formatString } from '@internal/acs-ui-common';
+import { useEffect, useState } from 'react';
+import { useLocale } from '../../localization';
+import { useAdapter } from '../adapter/CallAdapterProvider';
+
+/**
+ * Delay value for when a participant has left or joined
+ *
+ * this is to allow the narrator some room so it isnt stuttering whenever there is a change
+ * in participants.
+ */
+const PARTICIPANT_ANNOUNCEMENT_DELAY = 500;
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ *
+ * @internal
+ */
+export const useParticipantChangedAnnouncement = (): string => {
+  const adapter = useAdapter();
+  const locale = useLocale().strings.call;
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  /**
+   * state to track whether there is currently a timer set in the useParticipantChangedAnnouncement hook
+   */
+  const [timeoutState, setTimeoutState] = useState<ReturnType<typeof setTimeout>>();
+
+  useEffect(() => {
+    const setParticipantEventString = (string: string): void => {
+      setAnnouncerString('');
```

Comment:
what happens if names are being read out and a new user joins? Does it finish what it was saying then read out the new user or does it cut off the old announcement (potentially missing names) then read the new announcement?

Reply:
with polite being set it should finish the current name and then queue up another. but if someone joins in the window their announcement might be lost. I am still waiting to hear back what they are wanting us to do when this happens